src/rulesets/Base/fastmath_able.jl

```diff
@@ -37,9 +37,30 @@ let
 
         # Unary complex functions
         @scalar_rule abs(x::Real) sign(x)
-        @scalar_rule abs2(x) 2x
+        @scalar_rule abs2(x::Real) 2x
         @scalar_rule angle(x::Real) Zero()
         @scalar_rule conj(x::Real) One()
+        function frule((_, Δz), ::typeof(abs), z::Complex)
+            Ω = abs(z)
+            return Ω, (real(z) * real(Δz) + imag(z) * imag(Δz)) / Ω
+        end
+        function rrule(::typeof(abs), z::Complex)
+            Ω = abs(z)
+            function ∂abs(Δz)
+                return real(Δz)*conj(z)/Ω
```

Comment:
Need to also return the derivative with respect to the fields of function itself, which since it isn't a closure or callable struct is the `NO_FIELDS` constant (which is actually just `Zero`).

```suggestion
                return (NO_FIELDS, real(Δz)*conj(z)/Ω)
```